src/azul/http.py

```diff
@@ -386,3 +386,39 @@ def urlopen(self,
                         return response
             else:
                 return response
+
+
+def parse_header(header: str) -> tuple[str, dict[str, str]]:
+    """
+    Parse representation headers, like content-type, and return a key
+    consisting of the media type (MIME type) and the parameters/directives of
+    the specified media type.
+
+    >>> parse_header('text/html; charset=utf-8')
+    ('text/html', {'charset': 'utf-8'})
+
+    >>> key, params = parse_header('application/json; charset=utf-8; foo=bar')
+    >>> key
+    'application/json'
+    >>> params.items()
+    dict_items([('charset', 'utf-8'), ('foo', 'bar')])
+
+    >>> parse_header('text/html')
+    ('text/html', {})
+
+    >>> parse_header('')
+    ('', {})
+    """
+    from email.message import (
+        Message,
+    )
+    m = Message()
+    m.add_header('content-type', header)
+    key_content: list[tuple[str, str]] | None = m.get_params()
+    assert key_content is not None, R('Unparsable header', header)
```

Comment:
Should have doctests for these.

Reply:
Agreed during PL that this wasn't necessary for this assertion.